src/librustdoc/html/render/mod.rs

```diff
@@ -3693,7 +3711,9 @@ fn spotlight_decl(decl: &clean::FnDecl) -> String {
         if let Some(impls) = c.impls.get(&did) {
             for i in impls {
                 let impl_ = i.inner_impl();
-                if impl_.trait_.def_id().map_or(false, |d| c.traits[&d].is_spotlight) {
+                if impl_.trait_.def_id().map_or(false, |d| {
+                    clean::utils::has_doc_flag(c.tcx.as_ref().unwrap().get_attrs(d), sym::spotlight)
```

Comment:
As Joshua said in #80883, `TyCtxt` is Copy so you don't need `as_ref()`:

```suggestion
                    clean::utils::has_doc_flag(c.tcx.unwrap().get_attrs(d), sym::spotlight)
```

Reply:
Indeed!